lib/core/src/display/shape/text/text_field/keyboard.rs

```diff
@@ -0,0 +1,197 @@
+    /// A lambda node performing cut operation. Returns the string which should be copied to
+    /// clipboard.
+    cut_action: Dynamic<String>,
+    /// A lambda node performing copy operation. Returns the string which should be copied to
+    /// clipboard.
+    copy_action: Dynamic<String>,
+    /// A lambda node performing paste operation.
+    paste_action: Dynamic<()>,
+    /// A lambda node performing character input operation.
+    char_typed_action: Dynamic<()>,
+}
+
+impl TextFieldFrp {
+    /// Create FRP graph operating on given TextField pointer.
+    pub fn new(text_field_ptr:Weak<RefCell<TextFieldData>>) -> TextFieldFrp {
+        let keyboard          = Keyboard::default();
+        let mut actions       = KeyboardActions::new(&keyboard);
+        let cut_action        = Self::copy_action_lambda(true,text_field_ptr.clone());
+        let copy_action       = Self::copy_action_lambda(false,text_field_ptr.clone());
+        let paste_action      = Self::paste_action_lambda(text_field_ptr.clone());
+        let char_typed_action = Self::char_typed_lambda(text_field_ptr.clone());
+        frp! {
+            text_field.cut               = source();
+            text_field.copy              = source();
+            text_field.paste             = source();
+            text_field.copy_action       = copy .map(move |()| copy_action());
+            text_field.cut_action        = cut  .map(move |()| cut_action());
+            text_field.paste_action      = paste.map(paste_action);
+            text_field.char_typed_action = keyboard.key_pressed.map2(&keyboard.key_mask,char_typed_action);
+        }
+        Self::initialize_actions_map(&mut actions,text_field_ptr);
+        TextFieldFrp
+            {keyboard,actions,cut,copy,paste,cut_action,copy_action,paste_action,char_typed_action}
+    }
+
+    /// Bind this FRP graph to js events.
+    ///
+    /// Until the returned `KeyboardBinding` structure lives, the js events will emit the proper
+    /// source events in this graph.
+    pub fn bind_frp_to_js_text_input_actions(&self) -> KeyboardBinding {
+        let mut binding      = KeyboardBinding::create();
+        let frp_key_pressed  = self.keyboard.key_pressed.clone_ref();
+        let frp_key_released = self.keyboard.key_released.clone_ref();
+        let frp_cut          = self.cut.clone_ref();
+        let frp_copy         = self.copy.clone_ref();
+        let frp_paste        = self.paste.clone_ref();
+        let frp_text_to_copy = self.copy_action.clone_ref();
+        binding.set_key_down_handler(move |event:KeyboardEvent| {
+            if let Ok(key) = event.key().parse::<Key>() {
+                frp_key_pressed.event.emit(key);
+            }
+        });
+        binding.set_key_up_handler(move |event:KeyboardEvent| {
+            if let Ok(key) = event.key().parse::<Key>() {
+                frp_key_released.event.emit(key);
+            }
+        });
+        binding.set_copy_handler(move |is_cut| {
```

Comment:
All code regarding keyboard management (key down, key oup, and very common FRP operations, should be refactored out to our FRP library. The FRP library should provide us with a common Mouse / Keyboord operations AND utils to bind them to JS